core-bundle/src/Image/Studio/Figure.php

```diff
@@ -24,8 +24,10 @@
  * use the provided legacy helper methods to manually apply the data to them.
  *
  * Wherever possible, the actual data is only requested/built on demand.
+ *
+ * @final
```

Comment:
We are not officially supporting the `@final` annotation.

Reply:
Maybe we should? :smiley_cat:

Reply:
Yes, however, using `@final` did not have any effect in PhpStorm. I did not even get noticed that I am trying to extend a final class, so I guess the effect will be limited. 🤷🏻

Reply:
> Yes, however, using @final did not have any effect in PhpStorm.

:flushed: That's a bad thing indeed.

The annotation is still used a lot in Symfony and other projects, so I think devs are used to it.

Reply:
We could also use https://github.com/dg/bypass-finals (the lib is basically a stream wrapper)… not sure that's worth it, though.

Reply:
> Yes, however, using `@final` did not have any effect in PhpStorm.

I added a comment to https://youtrack.jetbrains.com/issue/WI-31225

Reply:
https://github.com/contao/contao/pull/3185